src/github.rs

```diff
@@ -3049,86 +3053,55 @@ async fn project_items_by_status(
     Ok(all_items)
 }
 
-/// Retrieve all pull requests in status OPEN that are not drafts
-pub async fn retrieve_open_pull_requests(
+/// Retrieve tuples of (user, PR number) where
+/// the given user is assigned as a reviewer for that PR.
+/// Only non-draft, non-rollup and open PRs are taken into account.
+pub async fn retrieve_pull_request_assignments(
     owner: &str,
     repository: &str,
-    client: &GithubClient,
-) -> anyhow::Result<Vec<(User, i32)>> {
-    use cynic::QueryBuilder;
-    use github_graphql::pull_requests_open::{PullRequestsOpen, PullRequestsOpenVariables};
-
-    let mut prs = vec![];
-
-    let mut vars = PullRequestsOpenVariables {
-        repo_owner: owner,
-        repo_name: repository,
-        after: None,
-    };
-    loop {
-        let query = PullRequestsOpen::build(vars.clone());
-        let req = client.post(&client.graphql_url);
-        let req = req.json(&query);
-
-        let data: cynic::GraphQlResponse<PullRequestsOpen> = client.json(req).await?;
-        if let Some(errors) = data.errors {
-            anyhow::bail!("There were graphql errors. {:?}", errors);
+    client: &Octocrab,
+) -> anyhow::Result<Vec<(User, PullRequestNumber)>> {
+    let mut assignments = vec![];
+
+    // We use the REST API to fetch open pull requests, as it is much (~5-10x)
+    // faster than using GraphQL here.
+    let stream = client
+        .pulls(owner, repository)
+        .list()
+        .state(State::Open)
+        .direction(Direction::Ascending)
+        .sort(Sort::Created)
+        .per_page(100)
+        .send()
+        .await?
+        .into_stream(client);
+    let mut stream = std::pin::pin!(stream);
+    while let Some(pr) = stream.try_next().await? {
+        if pr.draft == Some(true) {
+            continue;
         }
-        let repository = data
-            .data
-            .ok_or_else(|| anyhow::anyhow!("No data returned."))?
-            .repository
-            .ok_or_else(|| anyhow::anyhow!("No repository."))?;
-        prs.extend(repository.pull_requests.nodes);
-
-        let page_info = repository.pull_requests.page_info;
-        if !page_info.has_next_page || page_info.end_cursor.is_none() {
-            break;
+        // exclude rollup PRs
+        if pr
+            .labels
+            .unwrap_or_default()
+            .iter()
+            .any(|label| label.name == "rollup")
+        {
+            continue;
+        }
+        for user in pr.assignees.unwrap_or_default() {
+            assignments.push((
+                User {
+                    login: user.login,
+                    id: (*user.id).into(),
```

Comment:
This is changing `database_id` (only on GraphQl?) to `id (only on REST?). Are we sure this won't cause issue with existing GitHub calls?

Reply:
GitHub IDs are a bit confusing. My understanding is that GQL's `database_id` corresponds to the `id` used in most REST API calls.